### DISCUSSION TITLE
Cleanup Dependendy Dashboard automatically

### DISCUSSION THREAD

**Discussion by jgoux** [renovatebot/renovate#8754] [Request Help] [answered]
Hello all,

I've made a big refactor in our current repository to turn it into a monorepo.

I was wondering if renovate auto-clean the dependency dashboard on each run or if I have to do something about it?

Currently : 
![image](https://user-images.githubusercontent.com/1443499/108341496-fd831400-71d9-11eb-92b6-e7314562d830.png)

It's listing dependencies that are not there anymore such as is-ci or npm-check-updates. Also there are checkboxes for minor/patches modules. That's unexpected as I'm on automerge mode for these channels : 

```json
{
  "automerge": true,
  "extends": [
    "config:base"
  ],
  "major": {
    "automerge": false,
    "dependencyDashboardApproval": true
  },
  "packageRules": [
    {
      "matchPackageNames": ["parcel"],
      "allowedVersions": "!/nightly/"
    }
  ],
  "prCreation": "not-pending",
  "prHourlyLimit": 10,
  "schedule": [
    "after 9am and before 6pm every weekday"
  ],
  "stabilityDays": 3,
  "timezone": "Europe/Paris"
}

```

**Comment by rarkins** [comment]
It *should* clean these up. Can you tick the "re-run" on this repository, and wait for it to run (you should be able to tell because the checkbox gets unchecked once it's done). If it's still out of date, let's check the logs to see if the config is doing what you expect to be doing (e.g. check the list of branches in the logs).

**Reply by jgoux** [reply]
I checked the box and the bot ran a couple of times already. The box is still checked and the list isn't updated.
The last run is job number  #304085125 if it can help. 👍

**Comment by rarkins** [comment, marked as answer]
Please add `"dependencyDashboard": true` to the top level of your config. I think it's possible Renovate doesn't even know you have a dashboard so is neither reading from it or writing to it.

**Reply by jgoux** [reply]
That's it! The note at the end of the dependencyDashboardApproval confused me : 

> Note: Enabling Dependency Dashboard Approval implicitly enables dependencyDashboard too, so it is not necessary to configure both to true.

Because renovatebot did create the dependency dashboard without having `"dependencyDashboard": true` at the root. But it seems that it's ignoring it when doing its update job.

**Reply by rarkins** [reply]
Ideally we could traverse all though the code to find *any* reference to dependencyDashboard approval, or maybe we'd be better off just telling people to ensure they have the dashboard configured.
